lib/adapters/class-vip-enterprise-search.php

```diff
@@ -20,141 +17,112 @@
 class VIP_Enterprise_Search extends Adapter {
 
 	/**
-	 * Add facets to EP query.
-	 * Filters `ep_post_formatted_args`.
+	 * A callback for the ep_valid_response action hook. Parses aggregations
+	 * from the raw Elasticsearch response and adds the buckets to the
+	 * configured aggregations.
 	 *
-	 * @see \ElasticPress\Indexable\Post\Post::format_args() For the `ep_post_formatted_args` filter.
-	 *
-	 * @param array $query Formatted Elasticsearch query.
-	 * @return array
+	 * @param array $response Response from the Elasticsearch server.
 	 */
-	public function add_facets_to_ep_query( $query ): array {
-		// Do we have any facets specified?
-		$searched_facets = get_query_var( 'fs' );
-		if ( empty( $searched_facets ) ) {
-			return $query;
+	public function action__ep_valid_response( $response ): void {
+		if ( ! empty( $response['aggregations'] ) ) {
+			$this->parse_aggregations( $response['aggregations'] );
 		}
+	}
 
-		// Try to get the list of configured facets.
-		$configured_facets = $this->get_facet_config();
-		if ( empty( $configured_facets ) ) {
-			return $query;
+	/**
+	 * A callback for the ep_elasticpress_enabled filter hook. Overrides the
+	 * normal behavior for ElasticPress to determine if it is enabled to allow
+	 * for an empty search string, if allowable by the configuration on this
+	 * adapter.
+	 *
+	 * @param bool     $enabled  Whether ElasticPress is enabled for the query or not.
+	 * @param WP_Query $wp_query The WP_Query being examined.
+	 *
+	 * @return bool Whether ElasticPress should be active for the query or not.
+	 */
+	public function filter__ep_elasticpress_enabled( $enabled, $wp_query ) {
```

Comment:
Yeah, I'd advocate a1 here too, and that's what I've historically done. I don't think that we should be hooking the same action or filter in the same class at the same priority more than once, and if we need to hook it at different priorities, that typically doesn't pass the smell test (although could be due to weird behavior in a plugin that we have no other way of overcoming). In most situations we should be able to hook a single action or filter at a specific priority once per class and if we need to split out functionality to avoid the function getting too long, we can do that with private methods, or what-have-you.